mkosi/config.py

```diff
@@ -1817,13 +1819,8 @@ def cache_manifest(self) -> dict[str, Any]:
             "prepare_scripts": sorted(
                 base64.b64encode(script.read_bytes()).decode() for script in self.prepare_scripts
             ),
-            # We don't use the full path here since tests will often use temporary directories for the output
-            # directory which would trigger a rebuild every time.
-            "tools_tree": self.tools_tree.name if self.tools_tree else None,
-            "tools_tree_distribution": self.tools_tree_distribution,
-            "tools_tree_release": self.tools_tree_release,
-            "tools_tree_mirror": self.tools_tree_mirror,
-            "tools_tree_packages": sorted(self.tools_tree_packages),
+            # Statting the root directory of the tools tree isn't fool proof but should be good enough.
+            "tools_tree": [self.tools_tree, self.tools_tree.stat().st_mtime_ns] if self.tools_tree else None,
```

Comment:
Nit: 
```suggestion
            "tools_tree": [self.tools_tree, self.tools_tree.stat().st_mtime_ns] if self.tools_tree else [],
```
Would keep the type of the value the same whether `self.tools_tree` is true or not.